### PR TITLE
Replace libc with rustix in some backends

### PR DESCRIPTION
**Pull request by notgull**
Replaces `libc` with `rustix` in the `epoll`, event ports and `poll`-based backends. This ensures that `libc` is entirely bypassed when making system calls.

We can't do `kqueue` until bytecodealliance/rustix#605 is released.

Supersedes #91

### CODE REVIEW

**Comment by notgull** [comment]
...and since bytecodealliance/rustix#605 is released now, `kqueue` can be implemented with `rustix` now too! (Or almost all of it). Thanks, @sunfishcode!

File: Cargo.toml [hunk 1]
```diff
@@ -30,6 +30,7 @@ autocfg = "1"
 
 [target.'cfg(any(unix, target_os = "fuchsia", target_os = "vxworks"))'.dependencies]
 libc = "0.2.77"
```
Comment (taiki-e) [inline Cargo.toml:32]: Can we move this to dev-deps?

Comment (notgull) [reply Cargo.toml:32]: It's still used by the kqueue backend. I can make it kqueue only if you want.

**Review by taiki-e — APPROVED** [review]
LGTM

**Pull request merged**